### ISSUE TITLE
Network error: Tree topology, PullBlocksToTip issue

### ISSUE THREAD

**Issue by dkijania** [input-output-hk/jormungandr#745]
**Describe the bug**
For given topology:

![image](https://user-images.githubusercontent.com/20424186/64250037-65299e80-cf15-11e9-80e4-68917e02d699.png)

There is error visible in logs for Leader3,Leader2 and Leader 1:
```
[INFO ][Leader1@127.0.0.1:8003]: Sep 04 11:07:16.016 WARN error while streaming response: Error { code: NotFound, cause: StringError("none of the starting points are found in the blockchain") }, task: server, task: network
[INFO ][Leader1@127.0.0.1:8003]: Sep 04 11:07:16.018 WARN error while streaming response: Error { code: NotFound, cause: StringError("none of the starting points are found in the blockchain") }, task: server, task: network
[INFO ][Leader3@127.0.0.1:8001]: Sep 04 11:07:16.016 WARN PullBlocksToTip response stream failed: Error { code: NotFound, source: Status { code: NotFound, message: "not found" } }, node_id: 35891453800649389366359429047430392199, peer_addr: 127.0.0.1:12003, task: network
[INFO ][Leader1@127.0.0.1:8003]: Sep 04 11:07:16.020 WARN error while streaming response: Error { code: NotFound, cause: StringError("none of the starting points are found in the blockchain") }, node_id: 78576782392483556410171659317301957050, peer_addr: 127.0.0.1:12002, task: network
[INFO ][Leader2@127.0.0.1:8000]: Sep 04 11:07:16.018 WARN PullBlocksToTip response stream failed: Error { code: NotFound, source: Status { code: NotFound, message: "not found" } }, node_id: 35891453800649389366359429047430392199, peer_addr: 127.0.0.1:12003, task: network
[INFO ][Leader1@127.0.0.1:8003]: Sep 04 11:07:16.022 WARN UploadBlocks request failed: Error { code: Internal, source: Status { code: Internal, message: "h2 protocol error: protocol error: unexpected internal error encountered" } }, node_id: 78576782392483556410171659317301957050, peer_addr: 127.0.0.1:12002, task: network
```
[logs.txt](https://github.com/input-output-hk/jormungandr/files/3574310/logs.txt)

**Mandatory Information**
1. `jcli --full-version` output;
`jormungandr 0.3.3 (master-f39cb72, release, windows [x86_64]) - [rustc 1.36.0 (a53f9df32 2019-07-03)]`
2. `jormungandr --full-version` output:
`jormungandr 0.3.3 (master-f39cb72, release, windows [x86_64]) - [rustc 1.36.0 (a53f9df32 2019-07-03)]`
3. jormungandr-scenario-test seed:
`116ee1a7e5d6d0468ac20ab37e02bed5a32873fac2ac0fba22a443e8e9ad2688`

**To Reproduce**
1.  checkout branch https://github.com/input-output-hk/jormungandr/pull/744
2. run command: jormungandr-scenario-tests --root-dir T3005--document
3. Observe logs
4. Run jcli commands (get account/message logs) to check status of transaction

**Expected behavior**
No error in logs. All Nodes are in sync

**Comment by dkijania** [comment]
Warning: This issue is not 100% reproducible. My effectiveness is around 33%.

**Comment by mzabaluev** [comment]
Will try to repro after fixing the other problems encountered with the scenario.

**Comment by mzabaluev** [comment]
It seems that we just need a stop gap behavior for the case when no checkpoints sent by the initiator of blockchain sync are found in storage of the peer that will serve missing blocks. In this case, it should fall back to block 0 as the starting point. This can occur initially and in testing, but less likely in real networks.

**Comment by mzabaluev** [comment]
Should be fixed by #767 and #772.

**Comment by dkijania** [comment]
Original issue is fixed. However there is another one similar to https://github.com/input-output-hk/jormungandr/issues/742. Even if block was produced by some node (through election process) other nodes shows transaction as pending
